pkg/registry/core/seed/strategy.go

```diff
@@ -61,13 +62,47 @@ func (s Strategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 func (s Strategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
 	newSeed := obj.(*core.Seed)
 	oldSeed := old.(*core.Seed)
+	syncDependencyWatchdogSettings(newSeed)
 	newSeed.Status = oldSeed.Status
 
 	if !apiequality.Semantic.DeepEqual(oldSeed.Spec, newSeed.Spec) {
 		newSeed.Generation = oldSeed.Generation + 1
 	}
 }
 
+func syncDependencyWatchdogSettings(seed *core.Seed) {
+	if seed.Spec.Settings == nil || seed.Spec.Settings.DependencyWatchdog == nil {
+		return
+	}
+	// keeping the field `weeder` and `endpoint` in sync
+	// Case 1: If weeder is specified, endpoint isn't -> do nothing
+	// Case 2: If weeder isn't specified, endpoint is -> make weeder=endpoint
+	// Case 3: If both are specified, give preference to weeder field -> set endpoint=weeder
+	// Case 4: If both not specified, default weeder.enabled to true
+	if seed.Spec.Settings.DependencyWatchdog.Weeder == nil {
+		seed.Spec.Settings.DependencyWatchdog.Weeder = &core.SeedSettingDependencyWatchdogWeeder{Enabled: true}
+		if seed.Spec.Settings.DependencyWatchdog.Endpoint != nil {
+			seed.Spec.Settings.DependencyWatchdog.Weeder = &core.SeedSettingDependencyWatchdogWeeder{Enabled: seed.Spec.Settings.DependencyWatchdog.Endpoint.Enabled}
+		}
```

Comment:
If `Endpoint == nil` then we should also default it to `true` here. Otherwise, we only set `Weeder.enabled=true` while `Endpoint` is kept `nil`.
Perhaps you could tackle this by moving the 

```go
		seed.Spec.Settings.DependencyWatchdog.Endpoint = &core.SeedSettingDependencyWatchdogEndpoint{Enabled: seed.Spec.Settings.DependencyWatchdog.Weeder.Enabled}
```

in L88 part out of the `else` block.